app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -178,12 +178,14 @@ def validate(self, data):
         return data
 
     def update(self, experiment, data):
-        control_branch_data = data.pop("reference_branch", {})
-        treatment_branches_data = data.pop("treatment_branches", {})
-
         with transaction.atomic():
+            if set(data.keys()).intersection({"reference_branch", "treatment_branches"}):
```

Comment:
Hmm, does this transaction need to be started at all here, if these keys are absent? Or could `update` just bail out early with a `return experiment` if these are missing?

Reply:
Gotta call super

Reply:
Yeah, but I mean something like:

```python
control_branch_data = data.pop("reference_branch", None)
treatment_branches_data = data.pop("treatment_branches", None)

experiment = super().update(experiment, data)
if control_branch_data is None and treatment_branches_data is None:
    return experiment

with transaction.atomic():
    experiment.delete_branches()
    # etc...
```